src/core/lib/channel/channel_stack_builder.cc

```diff
@@ -289,7 +290,8 @@ grpc_error* grpc_channel_stack_builder_finish(
   size_t channel_stack_size = grpc_channel_stack_size(filters, num_filters);
 
   // allocate memory, with prefix_bytes followed by channel_stack_size
-  *result = gpr_zalloc(prefix_bytes + channel_stack_size);
+  *result = gpr_malloc(prefix_bytes + channel_stack_size);
+  reinterpret_cast<grpc_channel*>(*result)->channelz_channel.release();
```

Comment:
Seems a little odd to call `release()` here.  Wouldn't it be cleaner to just use placement-new to initialize the `RefCountedPtr<>`?  Or better yet, give `grpc_channel` its own ctor and use placement-new to initialize the whole thing.

Reply:
+1 grpc_channel would benefit from having a ctor

Reply:
Done.